rust/src/transformations/make_stable_expr/expr_lit/mod.rs

```diff
@@ -0,0 +1,50 @@
+use polars::prelude::*;
+use polars_plan::utils::expr_output_name;
+
+use crate::core::{ExprFunction, Function, MetricSpace, StabilityMap, Transformation};
+use crate::domains::{ExprDomain, LogicalPlanDomain, OuterMetric, SeriesDomain};
+use crate::error::Fallible;
+use crate::transformations::DatasetMetric;
+
+#[cfg(test)]
+mod test;
+
+/// Make a Transformation that returns a literal.
+///
+/// # Arguments
+/// * `input_domain` - Domain of the expression to be applied.
+/// * `input_metric` - The metric under which neighboring LazyFrames are compared.
+/// * `expr` - A literal expression.
+pub fn make_expr_lit<M: OuterMetric>(
+    input_domain: ExprDomain,
+    input_metric: M,
+    expr: Expr,
+) -> Fallible<Transformation<ExprDomain, ExprDomain, M, M>>
+where
+    M::InnerMetric: DatasetMetric,
+    M::Distance: Clone,
+    (ExprDomain, M): MetricSpace,
+{
+    let Expr::Literal(literal_value) = &expr else {
+        return fallible!(MakeTransformation, "Expected literal expression");
+    };
+
+    let name = expr_output_name(&expr)?;
+    let dtype = literal_value.get_datatype();
+    let mut series_domain = SeriesDomain::new_from_field(Field::new(name.as_ref(), dtype))?;
+    series_domain.nullable = false;
```

Comment:
Not particular to this PR, but how important is it to mark this as nullable = false? Should it be the default? Could it be moved into `new_from_field`?

Reply:
`new_from_field` should instantiate with the broadest possible interpretation of the domain. All you've got to work with is the name of the series and data type-- it could have nans or nulls, and is not bounded. 

More narrowly, in the setting of the literal transformation, we know the literal is not null. So we may as well tighten the output domain to exclude nulls. The fact that it's non-null will prove useful in the fill_null PR.

In the fill_nan PR, this is adjusted again to be even tighter-- if the literal is a non-nan float, the atom domain is adjusted to exclude nans.